backend/data_adapter.go

```diff
@@ -17,8 +19,30 @@ func newDataSDKAdapter(handler QueryDataHandler) *dataSDKAdapter {
 	}
 }
 
+func withHeaderMiddleware(ctx context.Context, headers http.Header) context.Context {
+	if len(headers) > 0 {
+		ctx = httpclient.WithContextualMiddleware(ctx,
+			httpclient.MiddlewareFunc(func(opts httpclient.Options, next http.RoundTripper) http.RoundTripper {
```

Comment:
Nit. I would consider moving this to the httpclient package to live together with the other middlewares.

Reply:
The problem I see with that is that we would need to export this like other middlewares when it's just an instance of them (which for the moment, is only relevant to this `backend` package).

I think it's better to leave it here unexported to avoid unnecessary breaking changes in the future but if you think otherwise let me know!